lumicks/pylake/confocal.py

```diff
@@ -136,16 +137,29 @@ def _ordered_axes(self):
     def _to_spatial(self, data):
         raise NotImplementedError
 
-    def _image(self, color):
-        if color not in self._cache:
-            photon_counts = getattr(self, f"{color}_photon_count").data
-            self._cache[color] = reconstruct_image_sum(photon_counts, self.infowave.data, self._shape)
-            self._cache[color] = self._to_spatial(self._cache[color])
-        return self._cache[color]
-
-    def _timestamps(self, sample_timestamps):
-        timestamps = reconstruct_image(sample_timestamps, self.infowave.data, self._shape, reduce=np.mean)
-        return self._to_spatial(timestamps)
+    @cachetools.cachedmethod(operator.attrgetter('_cache'))
```

Comment:
Consider this syntax instead. I find it a little bit more relatable. However, neither one is exactly intuitive. Either way is fine in the end.
```suggestion
    @cachetools.cachedmethod(lambda self: self._cache)
```